src/main/java/fr/skytasul/guardianbeam/Laser.java

```diff
@@ -462,6 +470,18 @@ public CrystalLaser(Location start, Location end, int duration, int distance) th
 			
 			destroyPackets = Packets.createPacketsRemoveEntities(crystalID);
 		}
+		protected void refreshPackets() throws ReflectiveOperationException {
+			fakeCrystalDataWatcher = Packets.createFakeDataWatcher();
+			Packets.setCrystalWatcher(fakeCrystalDataWatcher, end);
+			if (Packets.version < 17) {
+				createCrystalPacket = Packets.createPacketEntitySpawnNormal(start, Packets.crystalID, Packets.crystalType);
+			} else {
+				createCrystalPacket = Packets.createPacketEntitySpawnNormal(crystal);
+			}
+			metadataPacketCrystal = Packets.createPacketMetadata(crystalID, fakeCrystalDataWatcher);
+			
+			destroyPackets = Packets.createPacketsRemoveEntities(crystalID);
+		}
```

Comment:
Everything here does not need to be recreated each time, for instance the destroy packet.